backend/src/main/kotlin/co/chainring/core/utils/Utils.kt

```
@@ -11,3 +18,12 @@ fun generateHexString(length: Int = 64): String {
 fun ByteArray.toHex() = "0x" + this.toHexString(HexFormat.Default)
 
 fun String.toHexBytes() = Hex.decode(this.replace("0x", ""))
+
+fun LocalTime.truncateTo(unit: DateTimeUnit.TimeBased): LocalTime =
+    LocalTime.fromNanosecondOfDay(toNanosecondOfDay().let { it - it % unit.nanoseconds })
+
+fun LocalDateTime.truncateTo(unit: DateTimeUnit.TimeBased): LocalDateTime =
+    LocalDateTime(date, time.truncateTo(unit))
+
+fun Instant.truncateTo(unit: DateTimeUnit.TimeBased): Instant =
+    toLocalDateTime(TimeZone.UTC).truncateTo(unit).toInstant(TimeZone.UTC)
```

Seems like this could be replaced with something like this:

```
fun Instant.truncateTo(unit: DateTimeUnit.TimeBased): Instant =
    Instant.fromEpochSeconds(toEpochMilliseconds().let { it - it % unit.duration.inWholeMilliseconds })
```

Now `.truncateTo(DateTimeUnit.NANOSECOND)` wont have any effect, but I'm not sure if solution with local time is any different in this regard. We could throw an exception if provided unit is less than millisecond.

---

yeah, few conversions less. Now since we do not need to support milli truncation it is fine.